pkg/webservice/handlers_test.go

```diff
@@ -1317,6 +1353,110 @@ func TestSetLoggerLevel(t *testing.T) {
 	assert.Equal(t, rr.Code, http.StatusOK)
 }
 
+func TestSpecificUserAndGroupResourceUsage(t *testing.T) {
+	part := setup(t, configDefault, 1)
+	userManager := ugm.GetUserManager()
+	userManager.ClearUserTrackers()
+	userManager.ClearGroupTrackers()
+
+	// add 1 application
+	app := addAppWithUserGroup(t, "app-1", part, "root.default", false, security.UserGroup{
+		User:   "testuser",
+		Groups: []string{"testgroup"},
+	})
+	res := &si.Resource{
+		Resources: map[string]*si.Quantity{"vcore": {Value: 1}},
+	}
+	ask := objects.NewAllocationAskFromSI(&si.AllocationAsk{
+		ApplicationID:  "app-1",
+		PartitionName:  part.Name,
+		ResourceAsk:    res,
+		MaxAllocations: 1})
+	err := app.AddAllocationAsk(ask)
+	assert.NilError(t, err, "ask should have been added to app")
+
+	// add an alloc
+	uuid := "uuid-1"
+	allocInfo := objects.NewAllocation(uuid, "node-1", ask)
+	app.AddAllocation(allocInfo)
+	assert.Assert(t, app.IsStarting(), "Application did not return starting state after alloc: %s", app.CurrentState())
```

Comment:
Can we factor this out to a separate function. This is at least repeated twice and with the split of the test for user(s) and group(s) handlers it will be used many times more.

Reply:
Good point @wilfred-s , addressed in latest PR.